CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(sol2)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+include_directories(.)
+
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/Modules" "${CMAKE_MODULE_PATH}")
+
+set(LUA_VERSION "5.3" CACHE STRING "The version of Lua needed (5.1, 5.2, 5.3, LuaJIT)")
```

Comment:
Change `LUA_VERSION` -> `SOL_LUA_VERSION`. Don't mess with the names created by CMake itself.